File: src/programmers/level2/spy_42578/spy.py
clothes1 = [["yellowhat", "headgear"], ["bluesunglasses", "eyewear"], ["green_turban", "headgear"]]
clothes2 = [["crowmask", "face"], ["bluesunglasses", "face"], ["smoky_makeup", "face"]]
def solution(clothes):
    d = {}
    for i in clothes:
        if d.get(i[1]):
            d[i[1]].append(i[0])
        else:
            d[i[1]] = [i[0]]

    answer = 1
    for i in d.keys():
        answer *= (len(d[i])+1)
    return answer

print(solution(clothes2)-1)

File: src/programmers/level2/spy_42578/Spy.java
package programmers.level2.spy_42578;

import java.util.*;

public class Spy {
    public int solution(String[][] clothes){
        int answer = 1;
        Map<String, ArrayList<String>> m = new HashMap();
        for (String[] item : clothes) {
            if(m.containsKey(item[1])){
                m.get(item[1]).add(item[0]);
            }else{
                m.put(item[1], new ArrayList<>());
                m.get(item[1]).add(item[0]);
            }
        }

        Set<String> keys = m.keySet();
        for (String key : keys) {
            answer *= m.get(key).size() + 1;
        }

        return answer - 1;
    }

    public static void main(String[] args) {
        // test case
        String[][] clothes1 = {{"yellowhat", "headgear"}, {"bluesunglasses", "eyewear"}, {"green_turban", "headgear"}};
        String[][] clothes2 = {{"crowmask", "face"}, {"bluesunglasses", "face"}, {"smoky_makeup", "face"}};

        Spy s = new Spy();
        int result = s.solution(clothes2);
        System.out.println(result);
    }
}
